src/reducers/weatherReducer.js

```diff
@@ -0,0 +1,13 @@
+import { FETCH_WEATHER } from '../actions';
+
+const weatherReducer = (state = [], action) => {
+
+  switch (action.type) {
+    case FETCH_WEATHER:
+      return [action.payload, ...state];
```

Comment:
Very simple and nicely done - you could do your data manipulations here as well (as opposed to in your component), but I prefer a simple reducer.